### ISSUE TITLE
Use Win key as hotkey to trigger flow

### ISSUE THREAD

**Issue by jjw24** [Flow-Launcher/Flow.Launcher#662]
Using the Windows key to trigger flow, surely this can be done, just need to find the right win API library I suppose

**Comment by pc223** [comment]
I think many users like this feature, Flow can replace Windows search. I remember this was requested in Wox as well.

**Comment by jjw24** [comment]
Yeah it it's not doable due to the library we are using, we will need to find and switch to a different one.

**Comment by Megasteel32** [comment]
@jjw24 I use wox, considering switching due to an unadressed bug that renders the software useless. Regardless, I remaped the windows key to something else and wox to the windows key (or what it was) using some random key assignment software I found. was def easier than AHK

**Comment by jjw24** [comment]
> @jjw24 I use wox, considering switching due to an unadressed bug that renders the software useless. Regardless, I remaped the windows key to something else and wox to the windows key (or what it was) using some random key assignment software I found. was def easier than AHK

Yeah definitely tools out there that can map windows key, I just want to build it natively for flow.

**Comment by mon-jai** [comment]
I would like to have that too.

**Comment by jjw24** [comment]
For workarounds using other tools to get this working please see https://github.com/Flow-Launcher/Flow.Launcher/discussions/747

**Comment by Gurrrra** [comment]
> For workarounds using other tools to get this working please see #747

I tried this, but without all that extra since I just want to use win+space, but while Flow pops up when pressing those keys it seems that AHK keeps my ctrl button (since I use ctrl-space in Flow for the menu) is still pressed which makes it so I can't write anything unless I press ctrl first, then it works as it should.
I real fix for this would be awesome so we can set the win key as a trigger directly in Flow instead of using these janky workarounds that doesn't always work as they should :)

**Comment by jjw24** [comment]
There will be a new plugin coming to the store that will enable the ability to use the Win key to open and close flow, very good for those that are looking for this functionality. If the plugin hasn't shown up yet for you, you can download it from here also https://github.com/AminSallah/Flow.Launcher.Plugin.WinHotkey (`pm install <url_to_the_release_zip>`)

**Comment by NoPlagiarism** [comment]
Plugin released to store and supports custom shortcuts

**Comment by jjw24** [comment]
Working on this.

**Comment by brimwats** [comment]
very excited to hear progress is being made! I've made multiple attempts at using flow launcher over the years but I keep ending up using fluent or powertoys because of muscle memory!

**Comment by jjw24** [comment]
Please help test this new change:
- Search window can be triggered by the Windows key or up to four key combinations
- Customer Query Hotkeys can be triggered by the Windows key or up to four key combinations
- Other hotkeys such as for autocomplete or preview should still work as currently.

Download the portable version from the link below and copy your user data across (search `Flow Launcher Userdata`). I do not recommend replacing this as your main until it's available as a pre-release version because it may still contain some bugs.

https://ci.appveyor.com/api/buildjobs/lqj8vx1277c7fad7/artifacts/Output%2FPackages%2FFlow-Launcher-Portable.zip

**Comment by rottakore** [comment]
For the love of GOD, DO NOT USE AHK IN ANY SHAPE OR FORM to add this feature. AHK is detected by some anti-cheat systems as a cheat and can prevent you from even opening the game. Battlefield and The Finals are good examples. It even flagged the Win HotKey plugin from the Flow Launcher plugin store as a cheat when I used it. Because of this, I’m having to use [ATNSoft Key Remapper](https://atnsoft.com/keyremapper/) software to achieve this functionality.

**Comment by TheBestPessimist** [comment]
Maybe report this issue to those games via support tickets. AHK is most certainly a very useful tool. Heck, I choose to not play games which complain that I use AHK, instead of close AHK. It's that's useful for me.

**Comment by jjw24** [comment]
> DO NOT USE AHK to add this feature. AHK is detected by some anti-cheat systems as a cheat. Battlefield and The Finals are good examples. It even flagged the Win HotKey plugin from the Flow Launcher plugin store as a cheat when I used it. Because of this, I’m having to use [ATNSoft Key Remapper](https://atnsoft.com/keyremapper/) software to achieve this functionality.

It's not using AHK, it's using low level windows library to capture keys, but I don't know if it will get flagged as there is always a possibility being seen as key logging. Best is to test the above linked build and let me know the outcome.

**Comment by Nereithr** [comment]
> It's not using AHK, it's using low level windows library to capture keys, but I don't know if it will get flagged as there is always a possibility being seen as key logging. Best is to test the above linked build and let me know the outcome.

Working pretty great so far. Standard Win key shortcuts seem to be working, 

Some feature suggestions/issues I noticed to go along with this:

- [Issue] Modifier keys for this specific hotkey (didn't test others) don't work as expected:
  - If the modifier(s) is defined before the WIN key (i.e. the combo displays [SHIFT][WIN]) it just doesn't work
  - If the modifier(s) is defined after the WIN key, it only works when you first hold down the WIN key and then press the modifiers, the reverse doesn't work.
- [Issue] The hotkey swapper in the settings doesn't properly update the hotkey when rapidly switching between configurations. Quickly switch between multiple hotkey configurations, set it to, say, Win. It will work 90%, but sometimes it will act as if no hotkey is set and Win will open the Start menu
- [Feature/Issue] If the Flow Launcher Key is set to Win Logo with no modifier then the user can no longer access the regular Start Menu via hotkey, which may be desirable at times, namely to access the app grid:
  - Solutions 1/2: Since the Flow Launcher menu appears on key release, could add the following options:
    - Open Start Menu on doubletap (delay configurable/sane defaults/use the doubleclick delay). I like this because this is the default in GNOME linux
    - Open Start Menu after holding the key (delay configurable)
  - Solution 3: Shift/CTRL/ALT modifiers

**Comment by jjw24** [comment]
Win key to trigger flow is now enabled in the [prerelease](https://github.com/Flow-Launcher/Prereleases/releases/latest) and will be released in the upcoming v1.20.0.

Note:
Only the trigger flow hotkey is allowed to use Win key. Support for using Win key as part of a combination and using four-key combo hotkeys as mentioned in the previous comment is still WIP in PR 3157, because there are a few issues that came out of testing the PR that I need time to go through and resolve.

**Comment by Robert-K** [comment]
Loving the prerelease so far, amazing improvements across the board!

Unfortunately it seems the Win hotkey still has some issues:

Pressing Win shortly after logging in/unlocking the device (Win 11 26100) will not open Flow Launcher, but the Windows Start menu instead.

**Comment by Crunchbits** [comment]
The new built-in Win hotkey is working great for me, however, it is opening the Start Menu if a program that is running as Admin(such as Task Manager) is focused. If you run Flow Launcher as admin than the issue is fixed, however, as mentioned in [2639](https://github.com/Flow-Launcher/Flow.Launcher/issues/2639#issuecomment-2112117777) that could pose a security issue as any program then launched with Flow Launcher launches as administrator.

Hopefully there is some development to fix this without this security issue.

**Comment by Kacper-Kondracki** [comment]
I don't think Flow Launcher itself needs to run as administrator. I believe the best way to achieve Win Key hooking is indeed to use administrator privileges, but inside isolated process that does IPC with Flow Launcher
